source/openpulse/openpulse/parser.py

```diff
@@ -0,0 +1,264 @@
+"""
+=============================
+Parser (``openpulse.parser``)
+=============================
+
+Tools for parsing OpenPulse programs into the :obj:`reference AST <openpulse.ast>`.
+
+The quick-start interface is simply to call ``openpulse.parse``:
+
+.. currentmodule:: openpulse
+.. autofunction:: openpulse.parse
+
+The rest of this module provides some lower-level internals of the parser.
+
+.. currentmodule:: openpulse.parser
+.. autoclass:: OpenPulseNodeVisitor
+"""
+
+# pylint: disable=wrong-import-order
+
+__all__ = [
+    "parse",
+    "OpenPulseNodeVisitor",
+]
+
+from contextlib import contextmanager
+import re
+from typing import List
+
+try:
+    from antlr4 import CommonTokenStream, InputStream, ParserRuleContext
+except ImportError as exc:
+    raise ImportError(
+        "Parsing is not available unless the [parser] extra is installed,"
+        " such as by 'pip install openqasm3[parser]'."
+    ) from exc
+
+from openqasm3.parser import (
+    get_span,
+    add_span,
+    span,
+    QASMNodeVisitor,
+    _visit_identifier,
+    _raise_from_context,
```

Comment:
Should these methods be renamed to be public, ie., remove the snakecase?

Reply:
Just want to make sure I understand. These methods are imported from `openqasm3.parser`. Do you suggest that I modify `openqasm3.parser` to make them public?